### DISCUSSION TITLE
Convert to WebExtension!!

### DISCUSSION THREAD

**Discussion by black7375** [black7375/Firefox-UI-Fix#502] [Ideas]
I found out when I was trying to support [numirias/paxmod](https://github.com/numirias/paxmod) #500
We can switch the theme into a webextension and adjust the option in real time !!!

It will take time because I'm not used to the inside of Firefox, but I hope to be introduced in `V10.0.0`(Stage 3). #2

**Some Documents**

- [Welcome to WebExtensions Experiments](https://webextensions-experiments.readthedocs.io/en/latest/)
- [Extensions & Experiments](https://firefox-source-docs.mozilla.org/browser/urlbar/experiments.html)
- [WebExtensions API Development](https://firefox-source-docs.mozilla.org/toolkit/components/extensions/webextensions/index.html)

**Examples**
- https://github.com/numirias/paxmod
- https://github.com/mozilla/gecko-dev/tree/master/browser/extensions
- https://github.com/mozilla/gecko-dev/tree/master/browser/themes/addons

**Code**
- https://github.com/mozilla/gecko-dev/blob/f203d491e04dfbb0f70eb64217e4a6482f477ada/toolkit/modules/LightweightThemeConsumer.jsm#L407

**Comment by 117649** [comment]
@black7375 I have to point out 'experimental api' isn't WebExtension. It is a privileged tunnel to legacy granted to WebEx. I don't think it will work on release channel firefox (orange icon).

[vivaldi-fox](https://github.com/nt1m/vivaldi-fox) Is using real WebEx API but I'm afraid you will find its abilities are very limited.

It's kinda simple to achieve what you want with legacy. You just add an observer to the preference branch like this:

https://github.com/p1usminus/roomybookmarksredux/blob/f7643c0fa3a2286d4c4b78abbd22b75a5c804a45/src/content/overlay.js#L41-L46 

and let it reload style sheet whenever there is a change:

https://github.com/p1usminus/roomybookmarksredux/blob/f7643c0fa3a2286d4c4b78abbd22b75a5c804a45/src/content/overlay.js#L56-L62

All logic can be load to firefox in the form of a [userChromeJS](https://github.com/xiaoxiaoflood/firefox-scripts).

But I think the most simple solution is to load your css with StyloaiX and manually reload style whenever a change is made.
Although I haven't tested it yet.

**Reply by Aleksandra0056** [reply]
I agree with you, **it will not work on release channel with orange icon**. Paxmod is an extension which uses experimental APIs, and the Paxmod ISN'T working on release channel (because it uses experimental APIs).

**Reply by black7375** [reply]
That's a good point.

My plan is to [sign using `web-ext`](https://extensionworkshop.com/documentation/develop/getting-started-with-web-ext/).
In terms of [enpass's example](https://www.enpass.io/downloads/), the distribution itself seems to be possible without uploading to `addon.org`, and it is necessary to check whether it is possible when using `experimental api`.

**Reply by 117649** [reply]
> That's a good point.
> 
> My plan is to [sign using `web-ext`](https://extensionworkshop.com/documentation/develop/getting-started-with-web-ext/). In terms of [enpass's example](https://www.enpass.io/downloads/), the distribution itself seems to be possible without uploading to `addon.org`, and it is necessary to check whether it is possible when using `experimental api`.

All in all you have to use legacy anyway(use experimental api is use legacy). So why not just use legacy? And it is also working on orange icon.

**Comment by 117649** [comment]
@black7375 I've succeeded with StyloaiX. Reload style sheet can make option in effect on the fly.

**Comment by black7375** [comment]
This is an example of changing the browser UI using a web extension.

https://user-images.githubusercontent.com/25581533/235937256-762e901a-fe2c-49e3-ac34-ec691aa41908.mp4

**Reply by Aleksandra0056** [reply]
You have united back and forward buttons? But how? And it would be in next Lepton release?

**Reply by black7375** [reply]
No, There is no plan to change the default value.
And, it's already possible with the settings now.

**Reply by Aleksandra0056** [reply]
By the way, I no longer need these new buttons, because I've imported Strata back & forward buttons to my userChrome.css from CustomCSSforFx. But nice illustration video.

**Comment by Aleksandra0056** [comment]
I have one metaphor:
“God save the Fox, remove Chromium folks!”
